tests/standalone_tests/test_monet_db_facade.py

```diff
@@ -73,27 +82,44 @@ def test_broken_pipe_error_properly_handled(
 ):
     result = db_execute_and_fetchall(query="SELECT 1;")
     assert result[0][0] == 1
-    _restart_monetdb_container(MONETDB_LOCALNODETMP_NAME)
+    restart_monetdb_container(MONETDB_LOCALNODETMP_NAME)
+    db_execute_and_fetchall(query="SELECT 1;")
     result = db_execute_and_fetchall(query="SELECT 1;")
     out, err = capfd.readouterr()
-    assert "Trying to recover from BrokenPipeError" in err
+    print(f"{out=}")
+    print(f"{err=}")
+    assert (
+        "Trying to recover the connection with the database. Exception type: '<class 'BrokenPipeError'>'"
+        in err
+    )
     assert result[0][0] == 1
 
 
 @pytest.mark.slow
-def test_broken_pipe_error_properly_raised(
+def test_connection_refused_properly_handled(
+    capfd,
     monetdb_localnodetmp,
 ):
     result = db_execute_and_fetchall(query="SELECT 1;")
     assert result[0][0] == 1
-    _restart_monetdb_container(MONETDB_LOCALNODETMP_NAME)
-
-    with pytest.raises(BrokenPipeError), patch(
-        "mipengine.node.monetdb_interface.monet_db_facade.BROKEN_PIPE_MAX_ATTEMPTS", 1
-    ):
+    remove_monetdb_container(MONETDB_LOCALNODETMP_NAME)
+    with pytest.raises(ConnectionError):
         db_execute_and_fetchall(query="SELECT 1;")
+    out, err = capfd.readouterr()
+    assert (
+        "Trying to recover the connection with the database. Exception type: '<class 'OSError'>'"
+        in err
+    )
+
+
+def test_generic_exception_handled(monetdb_localnodetmp):
+    with pytest.raises(pymonetdb.exceptions.OperationalError):
+        db_execute_and_fetchall(query="SELECT * FROM non_existing_table;")
 
 
-def test_generic_exception_handled():
-    with pytest.raises(OSError):
+def test_connection_refused_with_stopped_monetdb_container(monetdb_localnodetmp):
```

Comment:
This is more or less a duplication of the `test_connection_refused_properly_handled`.

You can check here the scenario where monetdb is completely off.